OtripleS.Web.Api.Tests.Unit/Services/CourseAttachments/CourseAttachmentServiceTests.Exceptions.RetrieveById.cs

```diff
@@ -0,0 +1,169 @@
+        {
+            // given
+            Guid someAttachmentId = Guid.NewGuid();
+            Guid someCourseId = Guid.NewGuid();
+            SqlException sqlException = GetSqlException();
+
+            var expectedCourseAttachmentDependencyException = 
+                new CourseAttachmentDependencyException(sqlException);
+
+            this.storageBrokerMock.Setup(broker =>
+                 broker.SelectCourseAttachmentByIdAsync(someCourseId, someAttachmentId))
+                    .ThrowsAsync(sqlException);
+
+            // when
+            ValueTask<CourseAttachment> retrieveCourseAttachmentTask =
+                this.courseAttachmentService.RetrieveCourseAttachmentByIdAsync
+                    (someCourseId, someAttachmentId);
+
+            // then
+            await Assert.ThrowsAsync<CourseAttachmentDependencyException>(() =>
+                retrieveCourseAttachmentTask.AsTask());
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectCourseAttachmentByIdAsync(someCourseId, someAttachmentId),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogCritical(It.Is(SameExceptionAs(expectedCourseAttachmentDependencyException))),
+                    Times.Once);
+
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnRetrieveWhenDbExceptionOccursAndLogItAsync()
+        {
+            // given
+            Guid someAttachmentId = Guid.NewGuid();
+            Guid someCourseId = Guid.NewGuid();
+            var databaseUpdateException = new DbUpdateException();
+
+            var expectedCourseAttachmentDependencyException =
+                new CourseAttachmentDependencyException(databaseUpdateException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectCourseAttachmentByIdAsync(someCourseId, someAttachmentId))
+                    .ThrowsAsync(databaseUpdateException);
+
+            // when
+            ValueTask<CourseAttachment> retrieveAttachmentTask =
+                this.courseAttachmentService.RetrieveCourseAttachmentByIdAsync
+                    (someCourseId, someAttachmentId);
+
+            // then
+            await Assert.ThrowsAsync<CourseAttachmentDependencyException>(() =>
+                retrieveAttachmentTask.AsTask());
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectCourseAttachmentByIdAsync(someCourseId, someAttachmentId),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(expectedCourseAttachmentDependencyException))),
+                    Times.Once);
+
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
```

Comment:
Fix the order of the VNOs to match the order they are called in.
If they are never called they go to the bottom of the list